### ISSUE TITLE
Allowing an input field to change the Quality of Service parameter.

### ISSUE THREAD

**Issue by surfertas** [ros-visualization/rqt_image_view#41]
Hi all, thanks for maintaining this package.

I was wondering with ROS2 and the introduction of QoS, if it makes sense to allow for an input field that allows the user to change the QoS of the subscriber in order to meet the QoS compatibility [requirements](https://index.ros.org/doc/ros2/Concepts/About-Quality-of-Service-Settings/).

If this is something that is worth considering, I can try and work on a PR.

**Comment by BrettRD** [comment]
This is a standing problem, and I would love to see a PR for it.
The gazebo plugins seem to do their own thing, and cause a number of QoS issues.
Automatic selection of QoS, would help a lot.
related: https://github.com/ros-visualization/rqt/issues/187

**Comment by jacobperron** [comment]
FWIW, as of [ROS Galactic](https://index.ros.org/doc/ros2/Releases/Release-Galactic-Geochelone/#externally-configure-qos-at-start-up) we will be able to change the QoS settings via ROS parameters. Note, this require enabling the feature in rqt_image_view (which probably makes sense to do).

**Comment by karl-schulz** [comment]
Would it be an option to set the default QoS for subscribers in rqt_image_view to BEST_EFFORT? Most image data is published via BEST_EFFORT anyways, and messages with RELIABLE will still be shown (it's compatible in that direction).

Obviously, having an option like in RVIZ would be ideal, but probably also more work.

I posted this comment in https://github.com/ros-visualization/rqt/issues/187 first but I think is the right place.

**Comment by jacobperron** [comment]
@karl-schulz I think defaulting to "best effort" QoS makes sense for subscriptions. I'm happy to take a look at a PR changing the defaults.

**Comment by jacobperron** [comment]
Yes, please target `rolling-devel`. No need to touch the changelog file (it is updated during a release).

**Comment by karl-schulz** [comment]
Hi @jacobperron after some time I found the time to implement the change, but stumbled upon a different PR, using the QOS of a publisher on the given topic. I just wanted to make sure that we decide for the BEST_EFFORT default instead of this solution before creating a PR.

Issue:
https://github.com/ros-perception/image_pipeline/issues/158#issuecomment-157524991

Corresponding PR:
https://github.com/ros-visualization/rqt_image_view/pull/53

**Comment by jacobperron** [comment]
I think defaulting to best_effort (and providing the ability to override QoS options at start-up) is more straight-forward than adapting to the first discovered publisher (as in #53).
